### PR TITLE
Add slime filler unit test.

### PR DESCRIPTION
**Pull request by toregge**
@geirst : please review

### CODE REVIEW

File: searchsummary/src/tests/docsummary/slime_filler/slime_filler_test.cpp [hunk 1]
```diff
@@ -0,0 +1,623 @@
...
+}
+
+TEST_F(SlimeFillerTest, insert_array_filtered)
+{
+    auto array = make_array();
+    expect_insert_filtered(R"(["foo","bar","baz"])", array, {0, 1, 2});
+    expect_insert_filtered("null", array, {});
+    expect_insert_filtered(R"(["foo"])", array, {0});
+    expect_insert_filtered(R"(["bar"])", array, {1});
+    expect_insert_filtered(R"(["baz"])", array, {2});
```
Comment (geirst) [inline searchsummary/src/tests/docsummary/slime_filler/slime_filler_test.cpp:544]: Consider adding a single test case for 2 elements as well.

File: searchsummary/src/tests/docsummary/slime_filler/slime_filler_test.cpp [hunk 2]
```diff
@@ -0,0 +1,623 @@
...
+}
+
+TEST_F(SlimeFillerTest, insert_weighted_set_filtered)
+{
+    auto wset = make_weighted_set();
+    expect_insert_filtered(R"([{"item":"foo","weight":2},{"item":"bar","weight":4},{"item":"baz","weight":6}])", wset, {0, 1, 2});
+    expect_insert_filtered("null", wset, {});
+    expect_insert_filtered(R"([{"item":"foo","weight":2}])", wset, {0});
+    expect_insert_filtered(R"([{"item":"bar","weight":4}])", wset, {1});
+    expect_insert_filtered(R"([{"item":"baz","weight":6}])", wset, {2});
```
Comment (geirst) [inline searchsummary/src/tests/docsummary/slime_filler/slime_filler_test.cpp:561]: Consider adding a single test case for 2 elements as well.

File: searchsummary/src/tests/docsummary/slime_filler/slime_filler_test.cpp [hunk 3]
```diff
@@ -0,0 +1,623 @@
...
+}
+
+TEST_F(SlimeFillerTest, insert_map_filtered)
+{
+    auto map = make_map();
+    expect_insert_filtered(R"([{"key":"key1","value":"value1"},{"key":"key2","value":"value2"},{"key":"key3","value":"value3"}])", map, {0, 1, 2});
+    expect_insert_filtered("null", map, {});
+    expect_insert_filtered(R"([{"key":"key1","value":"value1"}])", map, {0});
+    expect_insert_filtered(R"([{"key":"key2","value":"value2"}])", map, {1});
+    expect_insert_filtered(R"([{"key":"key3","value":"value3"}])", map, {2});
```
Comment (geirst) [inline searchsummary/src/tests/docsummary/slime_filler/slime_filler_test.cpp:578]: Consider adding a single test case for 2 elements as well.

**Review by geirst — APPROVED** [review]
👍

File: searchsummary/src/tests/docsummary/slime_filler/slime_filler_test.cpp (hunk 1 above)
Comment (toregge) [reply searchsummary/src/tests/docsummary/slime_filler/slime_filler_test.cpp:544]: Done.

File: searchsummary/src/tests/docsummary/slime_filler/slime_filler_test.cpp (hunk 2 above)
Comment (toregge) [reply searchsummary/src/tests/docsummary/slime_filler/slime_filler_test.cpp:562]: Done.

File: searchsummary/src/tests/docsummary/slime_filler/slime_filler_test.cpp (hunk 3 above)
Comment (toregge) [reply searchsummary/src/tests/docsummary/slime_filler/slime_filler_test.cpp:580]: Done.

**Review by geirst — APPROVED** [review]
👍

**Pull request merged**